source/extensions/filters/http/transformation/transformation_filter.cc

```diff
@@ -205,6 +210,35 @@ void TransformationFilter::addEncoderData(Buffer::Instance &data) {
   encoder_callbacks_->addEncodedData(data, false);
 }
 
+void TransformationFilter::transformOnStreamCompletion() {
+  if (on_stream_completion_transformation_ == nullptr) {
+    return;
+  }
+
+  // Body isn't required for this transformer since it isn't included
+  // in access logs
+  Buffer::OwnedImpl emptyBody{};
+
+  // If response_headers_ is a nullptr (this can happpen if a client disconnects)
+  // we pass in an empty response header to avoid errors within the transformer.
+  if (response_headers_ == nullptr) {
+    response_headers_ = Http::ResponseHeaderMapImpl::create().get();
```

Comment:
this line is actually undefined behavior; you need to keep the unique_ptr for the duration the variable lives